### PR TITLE
Feature/ROC-7388 feature toggle for staff email settlement accepted

### PR DESCRIPTION
**Pull request by akritis22**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7388

### Change description ###
Added a feature toggle for staff email notification when the settlement is accepted

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5089163&bid=16742347)

File: src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    @BeforeEach
+    public void setup() {
+        event = new AgreementCountersignedEvent(SampleClaim.getWithSettlement(SampleSettlement.validDefaults()),
+            MadeBy.CLAIMANT,
+            "authorisation");
+    }
+
+    @Nested
+    @DisplayName("Staff email notification feature toggle test")
+    class StaffEmailNotificationFeatureToggle {
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java]: No need to nest these. If you just want the nice display name, you can put the annotation on the test methods themselves

File: src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import uk.gov.hmcts.cmc.domain.models.offers.MadeBy;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaim;
+import uk.gov.hmcts.cmc.domain.models.sampledata.offers.SampleSettlement;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class AgreementCountersignedStaffNotificationHandlerTest {
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java]: Can be package-private

File: src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Mock
+    private SettlementReachedStaffNotificationService settlementReachedStaffNotificationService;
+    private AgreementCountersignedStaffNotificationHandler handler;
+    private AgreementCountersignedEvent event;
+
+    @BeforeEach
+    public void setup() {
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java]: Can be package-private

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5089163&bid=16742347)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5089163&bid=16742347)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5089163&bid=16742347)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCountersignedStaffNotificationHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=5089163&bid=16742347)

**Pull request merged**